Update documentation with new authenticate usage on the CLI (#2865)
hashicorp/boundary: website/content/docs/oss/installing/run-and-login.mdx

Boundary uses a predictable login name (`admin`) and password (`password`) in
dev mode. These can be overridden, or randomly generated, with flags to

> `boundary dev`. The auth method resource resides in the global scope and is set
> as the primary for that scope, which means you don't need to specify the
> auth method ID as a flag when using the authenticate command, all you need
> is the short form `boundary authenticate`.

Reviewer: this could be written a little more clearly:

```suggestion
`boundary dev`. The auth method resource in the global scope is set as the primary auth method for that scope. This allows you to omit the auth method ID when authenticating, enabling the short-form `boundary authenticate`:
```

<Tabs>
<Tab heading="CLI">

```

> $ boundary authenticate 

Reviewer: same, I'd prefer to see output here

```suggestion
$ boundary authenticate
Please enter the login name (it will be hidden):
Please enter the password (it will be hidden):
```
